### ISSUE TITLE
Cannot catch login error (TypeError: statusMessage.replace is not a function)

### ISSUE THREAD

**Issue by Attacler** [Intevel/nuxt-directus#147]
### Version

nuxt-directus 5.2.1
nuxt: 3

### Steps to reproduce
Using the following code:
```ts
const email = ref(""),
  password = ref("");

const errorMessage = ref("");

function login() {
  const { login } = useDirectusAuth();

  login({
    email: email.value,
    password: password.value,
  })
    .then(() => {
      router.push("/dashboard");
    })
    .catch((error) => {
      console.log({ error });
      try {
        errorMessage.value = (error?.errors[0] as any).message;
      } catch (error) {
        errorMessage.value =
          "Unknown error occured, please try again or contact us.";
      }
    });
}
```
I run into the issue that the error is not getting passed trough correctly.
If i look into the {error} log, there is an issue about .replace is not a function.

### What is Expected?
Im expecting the first array of errors to show up

### What is actually happening?
It gives some kind of error (TypeError: statusMessage.replace is not a function)
![image](https://user-images.githubusercontent.com/42644889/236886428-6a95c572-00b1-453e-b38e-89d32e12d749.png)

**Comment by Intevel** [comment]
There was a breaking change with 

https://github.com/Intevel/nuxt-directus/commit/ac5fd4e76a6776afa8942a485c20b4eda41a5b6f.

**Comment by Attacler** [comment]
@Intevel the function does not return the errors array so i cannot do anything with it.

**Comment by Intevel** [comment]
You can parse it from the thrown error @Attacler

**Comment by Attacler** [comment]
> You can parse it from the thrown error @Attacler

Please explain that since the object/error only contains:
 - message: the error message, no array
 - stack: the error message, no array
 
 The array that you see ontop comes from the dev logging.

**Comment by Intevel** [comment]
I thought createError throws a new Error, but it doesn't. Thats why the try catch doesn't catch an error.
This is definetly a bug

**Comment by Attacler** [comment]
> I thought createError throws a new Error, but it doesn't. Thats why the try catch doesn't catch an error. This is definetly a bug

When do you think that this will be solved?
I have multiple projects depending on this package so it would be nice to show/capute the errors.

**Comment by Intevel** [comment]
In the next days, today or tomorrow.

**Comment by Sandros94** [comment]
While trying to create a reprod link for #148 I came across two more issues, this one and a directus one. So I decided to incorporate the reproduction for this one too.

The reproduction is part of the [directus branch in my test repo](https://github.com/Sandros94/test-nuxt3app/tree/directus), you could follow the steps below or the readme (I'll try to keep them in sync if I ever update one or the other).

## nuxt-directus
It is as simple as generating a `FORBIDDEN` error within directus and trying to access it with a fetch request from nuxt.

1. docker-compose up the `docker-compose-dev.yml`.
3. create a `links` collection (GUUID) with:
 - input: `name`
 - input: `link`
 - datepicker: `date`
 - statuses: `hidden` and `visible`
4. fill the collection with the provided `links.csv`.
5. apply a custom public access that filters only the `visible` items, **but don't grant any field permission** (intended).
 ![image.png](https://volta.s3.fr-par.scw.cloud/image_96a40a2a04.png)
6. start nuxt `npm i && npm run dev` and navigate to the links page.

This will generate the error in the webpage:
![image.png](https://volta.s3.fr-par.scw.cloud/image_d9f870e994.png)

But if we check the npm logs we see:
```log
[12:45:53] [Directus Error]: FetchError:  (403 Forbidden (http://localhost:8055/items/links?fields=id&fields=name&fields=link&fields=date&fields=status&filter=%7B%22status%22:%22visible%22%7D&sort=-date))
{  12:45:53  
  errors: [
    {
      message: "You don't have permission to access this.",
      extensions: [Object]
    }
}
[12:45:53] [Directus Error]: FetchError:  (403 Forbidden (http://localhost:8055/items/links?fields=id&fields=name&fields=link&fields=date&fields=status&filter=%7B%22status%22:%22visible%22%7D&sort=-date))
{                                                                                                                                                                     12:45:53
  errors: [
    {
      message: "You don't have permission to access this.",
      extensions: [Object]
    }
  ]
}
```

**Comment by Intevel** [comment]
Im actually trying to findout whats the best way to fix this. Is the problem that createError uses .replace?

**Comment by laportem** [comment]
any updates on a fix for this

**Comment by Intevel** [comment]
Im thinking about what is causing that? Should be the first error message of the response be the message of the createError?

**Comment by Sandros94** [comment]
> Im thinking about what is causing that? Should be the first error message of the response be the message of the createError?

Oh way, so this could happen because it is getting two errors at the same time?
I did notice it when I encountered it, but didn't much make sense since I (at least in my intention) did only 1 api call.

**Comment by Intevel** [comment]
Fixed in #150, release coming.
